modules/openldap/openldap.go

```diff
@@ -0,0 +1,123 @@
+package openldap
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultUser     = "admin"
+	defaultPassword = "adminpassword"
+	defaultRoot     = "dc=example,dc=org"
+	defaultAdminDn  = "cn=admin,dc=example,dc=org"
+)
+
+// OpenLDAPContainer represents the OpenLDAP container type used in the module
+type OpenLDAPContainer struct {
+	testcontainers.Container
+	adminUsername string
+	adminPassword string
+	rootDn        string
+}
+
+// ConnectionString returns the connection string for the OpenLDAP container
+func (c *OpenLDAPContainer) ConnectionString(ctx context.Context, args ...string) (string, error) {
+	containerPort, err := c.MappedPort(ctx, "1389/tcp")
+	if err != nil {
+		return "", err
+	}
+
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	connStr := fmt.Sprintf("ldap://%s", net.JoinHostPort(host, containerPort.Port()))
+	return connStr, nil
+}
+
+// LoadLdif loads an ldif file into the OpenLDAP container
+func (c *OpenLDAPContainer) LoadLdif(ctx context.Context, ldif []byte) error {
```

Comment:
Do we want this API method here, or as a functional option to be passed to the `RunContainer` function? If the latter, I'd go with adding a post-starts hook doing what you already did in this method

Reply:
@jespino I think you want to add this method also as part of the functional options. Is it fine if we merge the PR as is and that option comes as a follow-up? I'd like to have it as part of the PR, but I also see it important to value your work merging the PR now.

Reply:
I'm actually working on that right now, I'm struggling to find the way of doing it in a way that I feel comfortable with.

Reply:
Great then! Let me know if I can be of any help